### ISSUE TITLE
Pylint doesn't report logical tautologies or contradictions

### ISSUE THREAD

**Issue by bje-** [PyCQA/pylint#2051]
### Steps to reproduce

```
"""Test case"""

def example(arg):
    """An example."""
    if arg == arg:
        print("Yes")
```

### Current behavior

No warning.

### Expected behavior

I would expect Pylint to report that `arg == arg` is a logical tautology. Likewise if the code read `arg != arg`.

### pylint --version output

```
pylint 2.0.0
astroid 1.6.3
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
Thanks for the issue! Yes, pylint doesn't currently emit this error, but this check should be simple enough to implement.

**Comment by sushobhit27** [comment]
@PCManticore I was looking to fix this issue and was almost done with it but found things:
The checker which I implemented also reports message for existing test cases also, e.g
https://github.com/PyCQA/pylint/blob/1c0356f0683901e6e3561f4caeb176ff3e20be1e/pylint/test/functional/singleton_comparison.py#L7

https://github.com/PyCQA/pylint/blob/1c0356f0683901e6e3561f4caeb176ff3e20be1e/pylint/test/functional/too_many_boolean_expressions.py#L16

So shall this new checker only check for comparison b/w variables or literals also which my checker is currently doing?

**Comment by bje-** [comment]
My opinion is that literal coparisons should also be included in the check. Things that look fishy like this can sometimes point to real bugs (ie. the user meant to test a variable with `True`, but absent mindedly typed `True == True`).

**Comment by sushobhit27** [comment]
@PCManticore I have raised PR for this issue which checks literals also. Please review it.
